src/tools/rados/rados.cc

```diff
@@ -460,6 +461,53 @@ static int do_put(IoCtx& io_ctx, RadosStriper& striper,
   return ret;
 }
 
+static int do_append(IoCtx& io_ctx, RadosStriper& striper,
+                  const char *objname, const char *infile, int op_size,
+                  bool use_striper)
+{
+  string oid(objname);
+  bufferlist indata;
+  bool stdio = false;
+  if (strcmp(infile, "-") == 0)
+    stdio = true;
+
+  int ret;
+  int fd = STDIN_FILENO;
+  if (!stdio)
+    fd = open(infile, O_RDONLY);
+  if (fd < 0) {
+    cerr << "error reading input file " << infile << ": " << cpp_strerror(errno) << std::endl;
+    return 1;
+  }
+  char *buf = new char[op_size];
+  int count = op_size;
+  while (count != 0) {
+    count = read(fd, buf, op_size);
+    if (count < 0) {
+      ret = -errno;
```

Comment:
If this is read(2), you need to check for EINTR...

Yes, do_put also appears to have this bug.  I suggest you use safe_read from common/safe_io.h.